### PR TITLE
Add AssertResourceToClosedResourceRector rule

### PR DESCRIPTION
**Pull request by peter279k**
# Changed log

- It's related to this [PR](https://github.com/sebastianbergmann/phpunit/pull/4365).

### CODE REVIEW

**Comment by peter279k** [comment]
The CI has been failed on `composer phpstan` command running.

After checking these PHPStan static code analysis issues, it seems that these problematic codes are not related to mine.

The static code analysis messages are available [here](https://github.com/rectorphp/rector/pull/4037/checks?check_run_id=1035585875).

File: rules/phpunit/src/Rector/MethodCall/AssertResourceToClosedResourceRector.php [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+use PhpParser\Node;
+use PhpParser\Node\Expr\MethodCall;
+use Rector\Core\PhpParser\Node\Manipulator\IdentifierManipulator;
+use Rector\Core\Rector\AbstractPHPUnitRector;
+use Rector\Core\RectorDefinition\CodeSample;
+use Rector\Core\RectorDefinition\RectorDefinition;
+
+/**
+ * @see \Rector\PHPUnit\Tests\Rector\MethodCall\AssertResourceToClosedResourceRector\AssertResourceToClosedResourceRectorTest
```
[the remark this answers is not in the document]
Comment (TomasVotruba) [reply rules/phpunit/src/Rector/MethodCall/AssertResourceToClosedResourceRector.php:15]: Here is an example: https://github.com/rectorphp/rector/blob/dffcae0c82cdc484aa2a46da94ea4004aee313f0/rules/php74/src/Rector/Property/TypedPropertyRector.php#L32

The purpose is to explain "why" for the existance of the rule to the future generation (that will try to delete everything that is not useful anymore :))

**Comment by TomasVotruba** [comment]
Could you rebase on `master`? The PHPStan errors should be fixed now

**Comment by peter279k** [comment]
It seems that the PHPStan static code analysis still reports error :(.

**Comment by peter279k** [comment]
It seems to be my static code error:

```
 rules/phpunit/src/Rector/MethodCall/AssertResourceToClosedResourceRector.php:17
Script vendor/bin/phpstan analyse --ansi --error-format symplify handling the phpstan event returned with error code 1
 ------------------------------------------------------------------------
 - '#Class "Rector\\PHPUnit\\Rector\\MethodCall\\AssertResourceToClosedResourceRector" is missing @see annotation with test case class reference#'
```

**Comment by peter279k** [comment]
Thanks, @TomasVotruba. I appreciate your help and review :)!

**Comment by TomasVotruba** [comment]
You're welcome.

Thank you for contribution :+1:

**Pull request merged**